### PR TITLE
Fix alert rules lost after rolling upgrade

### PR DESCRIPTION
**Pull request by weekface**
### What problem does this PR solve? 

Part of: https://github.com/pingcap/tidb-operator/issues/2710

### Check List 

Tests 

 - Unit test
 - E2E test
 - Stability test
 - Manual test (add detailed scripts or steps below)
 - No code

Code changes

 - Has Go code change
 - Has CI related scripts change
 - Has Terraform scripts change

Side effects

 - Breaking backward compatibility

Related changes

 - Need to cherry-pick to the release branch
 - Need to update the documentation

### Does this PR introduce a user-facing change?:

```release-note
Fix alert rules lost after rolling upgrade
```

### CODE REVIEW

File: charts/tidb-cluster/templates/monitor-deployment.yaml [hunk 1]
```diff
@@ -67,7 +67,11 @@ spec:
         - name: PROM_PERSISTENT_DIR
           value: /data
         - name: TIDB_VERSION
+      {{- if .Values.monitor.prometheus.alertManagerRulesVersion }}
+          value: tidb:{{ .Values.monitor.prometheus.alertManagerRulesVersion }}
```
Comment (weekface) [inline charts/tidb-cluster/templates/monitor-deployment.yaml:71]: Required format of `TIDB_VERSION` is `tidb:vx.x.x`.

**Comment by Yisaer** [comment]
Does `TidbMonitor` need this change?

**Comment by weekface** [comment]
> Does `TidbMonitor` need this change?

Yes. Another PR is coming soon which should only be cherry-picked to `release-1.1`.

**Pull request merged**